test/e2e/cni.go

```diff
@@ -0,0 +1,109 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	helmVals "helm.sh/helm/v3/pkg/cli/values"
+	k8snet "k8s.io/utils/net"
+	capi_e2e "sigs.k8s.io/cluster-api/test/e2e"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+)
+
+const (
+	calicoHelmChartRepoURL   string = "https://projectcalico.docs.tigera.io/charts"
+	calicoOperatorNamespace  string = "tigera-operator"
+	CalicoSystemNamespace    string = "calico-system"
+	CalicoAPIServerNamespace string = "calico-apiserver"
+	calicoHelmReleaseName    string = "projectcalico"
+	calicoHelmChartName      string = "tigera-operator"
+	ipv4                     string = "ipv4"
+	ipv6                     string = "ipv6"
+	dualStack                string = "dual-stack"
+)
+
+// InstallCalicoHelmChart installs the official calico helm chart
+// and validates that expected pods exist and are Ready.
+func InstallCalicoHelmChart(ctx context.Context, input clusterctl.ApplyClusterTemplateAndWaitInput, cidrBlocks []string) {
+	specName := "calico-install"
+
+	By("Installing Calico CNI via helm")
+	values := getCalicoValues(cidrBlocks)
+	InstallHelmChart(ctx, input, calicoOperatorNamespace, calicoHelmChartRepoURL, calicoHelmChartName, calicoHelmReleaseName, values)
+	clusterProxy := input.ClusterProxy.GetWorkloadCluster(ctx, input.ConfigCluster.Namespace, input.ConfigCluster.ClusterName)
+	workloadClusterClient := clusterProxy.GetClient()
+
+	By("Waiting for Ready tigera-operator deployment pods")
+	for _, d := range []string{"tigera-operator"} {
+		waitInput := GetWaitForDeploymentsAvailableInput(ctx, clusterProxy, d, calicoOperatorNamespace, specName)
+		WaitForDeploymentsAvailable(ctx, waitInput, e2eConfig.GetIntervals(specName, "wait-deployment")...)
+	}
+	By("Waiting for Ready calico-system deployment pods")
+	for _, d := range []string{"calico-kube-controllers", "calico-typha"} {
+		waitInput := GetWaitForDeploymentsAvailableInput(ctx, clusterProxy, d, CalicoSystemNamespace, specName)
+		WaitForDeploymentsAvailable(ctx, waitInput, e2eConfig.GetIntervals(specName, "wait-deployment")...)
+	}
+	By("Waiting for Ready calico-apiserver deployment pods")
+	for _, d := range []string{"calico-apiserver"} {
+		waitInput := GetWaitForDeploymentsAvailableInput(ctx, clusterProxy, d, CalicoAPIServerNamespace, specName)
+		WaitForDeploymentsAvailable(ctx, waitInput, e2eConfig.GetIntervals(specName, "wait-deployment")...)
+	}
+	By("Waiting for Ready calico-node daemonset pods")
+	for _, ds := range []string{"calico-node"} {
+		WaitForDaemonset(ctx, input, workloadClusterClient, ds, CalicoSystemNamespace)
+	}
+}
+
+func getCalicoValues(cidrBlocks []string) *helmVals.Options {
+	var ipv6, ipv4 string
```

Comment:
It looks like this would overshadow the const `ipv4` and `ipv6` declared at the top of the file and are used to specify CIDR blocks.
```suggestion
	var ipv6CidrBlock, ipv4CidrBlock string
```